components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -169,8 +169,44 @@ public User createUser(User user, boolean isBulkUserAdd) throws CharonException,
                 if (claimsMap.containsKey(SCIMConstants.USER_NAME_URI)) {
                     claimsMap.remove(SCIMConstants.USER_NAME_URI);
                 }
-                carbonUM.addUser(user.getUserName(), user.getPassword(), null, claimsMap, null);
+
+                if (claimsMap.containsKey(SCIMConstants.META_LOCATION_URI)) {
+                    claimsMap.remove(SCIMConstants.META_LOCATION_URI);
+                }
+
+                Map<String, String> clonedClaimsMap = new HashMap(claimsMap);
+                carbonUM.addUser(user.getUserName(), user.getPassword(), null, clonedClaimsMap, null);
+                //Remove username from the modified map before compare with the original map
+                String modifiedUserName = null;
+                if (clonedClaimsMap.containsKey(SCIMConstants.USER_NAME_URI)) {
+                    modifiedUserName = clonedClaimsMap.get(SCIMConstants.USER_NAME_URI);
+                    clonedClaimsMap.remove(SCIMConstants.USER_NAME_URI);
+                }
                 log.info("User: " + user.getUserName() + " is created through SCIM.");
+
+                // Check if the user claims map passed has been modified.
+                if (!claimsMap.keySet().containsAll(clonedClaimsMap.keySet()) || !clonedClaimsMap.keySet()
+                        .containsAll(claimsMap.keySet()) || !claimsMap.values().containsAll(clonedClaimsMap.values())) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Claims of user : " + user.getUserName() + " is updated. Populate updated claims.");
+                    }
+
+                    user.setUserName(modifiedUserName);
+
+                    clonedClaimsMap.put(SCIMConstants.USER_NAME_URI, user.getUserName());
+                    String userId = clonedClaimsMap.get(SCIMConstants.ID_URI);
+                    clonedClaimsMap.put(SCIMConstants.META_LOCATION_URI, SCIMCommonUtils.getSCIMUserURL(userId));
+
+                    //construct the SCIM Object from the attributes
+                    try {
+                        User newUser = (User) AttributeMapper
+                                .constructSCIMObjectFromAttributes(clonedClaimsMap, SCIMConstants.USER_INT);
+                        return newUser;
+                    } catch (NotFoundException e) {
+                        log.warn("Failed to populate modified claims for user : " + user.getUserName() + " created.",
```

Comment:
throw an exception, rather logging

Reply:
This is an API change and this should specifically handle in the charon level.